packages/mobile/src/identity/contactMapping.ts

```diff
@@ -218,19 +221,44 @@ export function* fetchAddressesAndValidateSaga({
   }
 }
 
-function* getAddresses(e164Number: string) {
-  let phoneHash: string
-  if (features.USE_PHONE_NUMBER_PRIVACY) {
-    const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
-    phoneHash = phoneHashDetails.phoneHash
-  } else {
-    phoneHash = getPhoneHash(e164Number)
-  }
-
+function* getAccountAddresses(e164Number: string) {
+  const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
+  const phoneHash = phoneHashDetails.phoneHash
   const lookupResult: IdentifierLookupResult = yield call(lookupAttestationIdentifiers, [phoneHash])
   return getAddressesFromLookupResult(lookupResult, phoneHash) || []
 }
 
+function* getWalletAddressesAndUpdateCache(e164Number: string) {
+  const contractKit = yield call(getContractKit)
+  const accountsWrapper: AccountsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAccounts,
+  ])
+
+  const accountAddresses: string[] = yield call(getAccountAddresses, e164Number)
+  const walletAddresses: string[] = yield all(
+    accountAddresses.map((accountAddress) => call(accountsWrapper.getWalletAddress, accountAddress))
+  )
+
+  const possibleUserAddresses: string[] = []
+  const walletToAccountAddress: WalletToAccountAddressType = {}
+  walletAddresses.forEach((address, i) => {
+    const walletAddress = address.toLowerCase()
+    const accountAddress = accountAddresses[i]
+    // `getWalletAddress` returns 0x0 when there isn't a wallet registered
+    if (walletAddress !== '0x0') {
+      walletToAccountAddress[walletAddress] = accountAddress
+      possibleUserAddresses.push(walletAddress)
```

Comment:
So the intention is to give precedence to MTWs instead of EOAs when we find one, is that right?

Reply:
With feeless attestations we are moving towards a world where the MTW is what gets verified instead of the EOA. That means when we check for addresses verified against a given identifier, we will be receiving back verified MTWs (for users who verified feelessly) and EOAs (for users who paid for their own verification).

What I am doing here is checking all the "verified" addresses and seeing if they have a `walletAddress` registered to them. This way, I can be sure that we will always be sending to EOAs (and never MTWs).

In the event they don't have a wallet registered (the `else` block), I am assuming that the "verified" address is an EOA. This mostly exists for the low probability edge case that a user has a verified account but has not yet registered their DEK on-chain.